amanda_keogh/test/test.js

```diff
@@ -0,0 +1,44 @@
+var chai = require('chai');
+var chaihttp = require('chai-http');
+chai.use(chaihttp);
+var expect = chai.expect;
+var fs = require('fs');
+require(__dirname + "/../server.js");
+
+describe('the express server', function() {
+  before(function() {
+    this.timestamp = Date.now();
+  });
+
+  after(function() {
+    fs.unlink(__dirname + '/../data/' + this.timestamp + '.json', function(err) {
+      if (err) throw err;
+    }.bind(this));
+  });
+
+  it('should write a file to the data directory', function(done) {
+    var directoryBefore = fs.readdirSync(__dirname + '/../data').length;
+    chai.request('localhost:3000')
+      .post('/' + this.timestamp)
+      .send({note: 'test note'})
+      .end(function(err, res) {
+        expect(err).to.eql(null);
+        expect(fs.readdirSync(__dirname + '/../data').length).to.eql(directoryBefore + 1);
+        expect(JSON.parse(fs.readFileSync(__dirname + '/../data/' + this.timestamp + '.json'))).to.eql({note: 'test note'});
+        done();
+      }.bind(this));
+  });
+
```

Comment:
Your second test is dependent on your first test, which isn't ideal. Try to make your unit tests able to run independently of each other.